mlos_bench/mlos_bench/services/remote/azure/azure_saas.py

```diff
@@ -0,0 +1,279 @@
+            vm_name="{vm_name}",
+            server_type="flexibleServers" if is_flex else "servers",
+            update="configurations",
+            api_version=api_version,
+        )
+
+        # These parameters can come from command line as strings, so conversion is needed.
+        self._request_timeout = float(self.config.get("requestTimeout", self._REQUEST_TIMEOUT))
+
+    def configure(self, config: Dict[str, Any],
+                  params: Dict[str, Any]) -> Tuple[Status, dict]:
+        """
+        Update the parameters of an Azure DB service.
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Key/value pairs of configuration parameters (e.g., vmName).
+        params : Dict[str, Any]
+            Key/value pairs of the service parameters to update.
+
+        Returns
+        -------
+        result : (Status, dict={})
+            A pair of Status and result. The result is always {}.
+            Status is one of {PENDING, SUCCEEDED, FAILED}
+        """
+        if self._is_batch:
+            return self._config_batch(config, params)
+        return self._config_many(config, params)
+
+    def is_config_pending_restart(self, config: Dict[str, Any]) -> Tuple[Status, dict]:
+        """
+        Check if the configuration of an Azure DB service requires a restart.
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Key/value pairs of configuration parameters (e.g., vmName).
+
+        Returns
+        -------
+        result : (Status, dict={"isConfigPendingRestart": bool})
+            A pair of Status and result. A Boolean field
+            "isConfigPendingRestart" indicates whether restart is required.
+            Status is one of {PENDING, TIMED_OUT, SUCCEEDED, FAILED}
+        """
+        config = merge_parameters(
+            dest=self.config.copy(), source=config, required_keys=["vmName"])
+        url = self._url_config_get.format(vm_name=config["vmName"])
+        _LOG.debug("Request: GET %s", url)
+        response = requests.put(
+            url, headers=self._get_headers(), timeout=self._request_timeout)
+        _LOG.debug("Response: %s :: %s", response, response.text)
+        if response.status_code == 504:
+            return (Status.TIMED_OUT, {})
+        if response.status_code != 200:
+            return (Status.FAILED, {})
+        return (Status.SUCCEEDED, {"isConfigPendingRestart": any(
+            {'False': False, 'True': True}[val['properties']['isConfigPendingRestart']]
+            for val in response.json()['value']
+        )})
+
+    def _get_headers(self) -> dict:
+        """
+        Get the headers for the REST API calls.
+        """
+        assert self._parent is not None and isinstance(self._parent, SupportsAuth), \
+            "Authorization service not provided. Include service-auth.jsonc?"
+
+        return {"Authorization": "Bearer " + self._parent.get_access_token()}
```

Comment:
Probably a future PR, but can we abstract that somehow for all AzureServices?  Or make it part of AzureAuth or something? It seems like we have some redundancy for all of our Azure interactions.

Reply:
great idea - I'll do it in the next PR

Reply:
File an issue for now?

Reply:
done: #581

Reply:
Also, using this method here

P.S. We cannot have that assertion in the constructor, because it will fail our unit tests in `test_load_service_config_examples.py`